htlcswitch/link_test.go

```diff
@@ -5444,13 +5483,16 @@ func TestChannelLinkFail(t *testing.T) {
 		// Execute the test case.
 		test.linkTest(t, coreLink, remoteChannel)
 
-		// Currently we expect all test cases to lead to link error.
 		var linkErr LinkFailureError
-		select {
-		case linkErr = <-linkErrors:
-		case <-time.After(10 * time.Second):
-			t.Fatalf("%d) Alice did not fail"+
-				"channel", i)
+		if test.shouldFailChannel {
```

Comment:
This doesn't test the reverse. If the channel is failed although this is unexpected, the test should fail.

Reply:
totally right, i added the new test case, saw it fail (as it should) and fix it this way but in reality I wasn't testing anything :-1: 

Fixed in both ways with

```
if test.shouldFailChannel && timeout {
	require.Fail(t, "expected link errors did not happen "+
		"before timeout")
} else if !test.shouldFailChannel && !timeout {
	require.Fail(t, "unexpected link errors happened")
}

```